WcaOnRails/spec/factories/persons.rb

```diff
@@ -4,7 +4,10 @@
     wca_id do
       mid = ('A'..'Z').to_a.sample(4).join
       "2016#{mid}01".tap do |id|
-        id.next! while Person.exists?(wca_id: id)
+        while Person.exists?(wca_id: id)
+          id = id.next
+        end
+        id
```

Comment:
Ah, good catch! If we don't mutate the string, then this `tap` is pointless. I've rewritten this.